library/std/src/sync/once_lock.rs

```diff
@@ -8,47 +8,17 @@ use crate::sync::Once;
 /// A synchronization primitive which can be written to only once.
 ///
 /// This type is a thread-safe [`OnceCell`], and can be used in statics.
+/// In many simple cases, you can use [`LazyLock<T, F>`] instead to get the benefits of this type
+/// with less effort: `LazyLock<T, F>` "looks like" `&T` because it initializes with `F` on deref!
+/// Where OnceLock shines is when LazyLock is too simple to support a given case, as LazyLock
+/// doesn't allow additional inputs to its function after you call [`LazyLock::new(|| ...)`].
```

Comment:
This sounds—at least for me—a bit misleading. I would write something like: "doesn't allow different initialization functions after you call ...". What do you think?

Reply:
@kaffarell The function is nullary.

Reply:
Well, OnceLock is also nullary, isn't it?

Reply:
Any function with N arguments can be curried to a nullary function, but this is not experienced in a profoundly different way than "calling the function". Yes, it is technically a "different function" because you enclose the state, but most people would describe this:
```rust
let clos = || f(a, b, c, d, etc);
let value = ONCE_LOCK.get_or_init(clos);
```
as calling `f`, instead of calling `clos`, because they actually write this:
```rust
let value = ONCE_LOCK.get_or_init(|| f(a, b, c, d, etc));
```